Comments block tweaks

bluefuton opened the pull request:
As per @fraying's issue #7423, this PR will make comments delightful in the following ways:

- [x] Add header showing total number of comments
- [x] Fix author link colours
- ~~Show site title where available (have to check if we have this information)~~ for a later PR
- [x] If the commenter has a URL, the gravatar should be linked, too.
- [x] Remove extra space above Reply and Like
- [x] Remove "open in new window" icon
- [x] Links on the commenter's name and site to open their feed stream in Reader
- [x] Drop date and time to new line

Fixes #7423.

Test live: https://calypso.live/?branch=update/blocks/comments-layout

bluefuton commented:
Adding the site information might be one for another PR. We don't currently appear to receive any commenter site information from the API.

fraying commented:
That’s cool. Listing the site is icing.

bluefuton commented:
@fraying  when we have a lot of comments to show, we show a link to load earlier comments. Would you like to do any tweaking here? This is what I've got at the moment:

<img width="754" alt="screen shot 2016-08-24 at 17 36 13" src="https://cloud.githubusercontent.com/assets/17325/17937163/7eba2966-6a21-11e6-8e1f-1dd82380257b.png">

For example: http://calypso.localhost:3000/read/feeds/40474296/posts/999143144

bluefuton commented:
Does the 'no comments' default feel a bit shouty?

<img width="765" alt="screen shot 2016-08-24 at 17 57 23" src="https://cloud.githubusercontent.com/assets/17325/17937969/4b9aa17a-6a24-11e6-912c-3c14b7047fb3.png">

jancavan commented on client/blocks/comments/style.scss:
```diff
@@ -350,3 +354,12 @@
+
+.comments__comment-count {
+	color: lighten( $gray, 10% );
+	size: 13px;
```
Is this supposed to be `font-size`? I think `13px` looks much better though. The original `17px` is a bit much in my opinion.

bluefuton replied:
Ah, good catch :) I thought it looked a bit big for 13px...

bluefuton replied:
I've set it to 14px as that's what @fraying originally specced in #7423. That's also the same size as the 'view earlier comments' link.

jancavan commented:
> Does the 'no comments' default feel a bit shouty?

I think mainly also because it looks large. "A" in add should also be capitalized.

![screenshot 2016-08-24 09 51 43](https://cloud.githubusercontent.com/assets/4924246/17939651/71ee04b4-69e0-11e6-9adb-2b25fdd46fb6.png)

fraying commented:
Good catch on the "A", @jancavan!

It feels a bit shouty because the other allcaps headers aren't in place yet. Once they're in, it'll blend right in.

fraying commented:
@bluefuton Thanks for the ping on the "View earlier comments" link. I think it's fine for now.

jancavan commented:
For Author names, links and non-links are all the same blue. It's not obvious which ones are links. I think for non-links, `darken( $gray, 30% )` works better.

fraying commented:
Good call, @jancavan. 👍

bluefuton commented:
Thanks both! I'll follow up on your feedback.

> "A" in add should also be capitalized.

Wearing my pedantic hat, I'd say we should either replace the dash or have a lower-case "a". The first word following a dash isn't usually capitalized:

http://www.quickanddirtytips.com/education/grammar/dashes-colons-and-commas

fraying commented:
It's true. In this case, the second part ("Add the first!") should really on its own line, below the "NO COMMENTS" header. That doable? If so, it should have the same font settings as the text in a comment.

Or we can leave it like it is.

jancavan commented:
> The first word following a dash isn't usually capitalized:

Ah, I thought there were exceptions, but TIL 😄

bluefuton commented:
I reckon 'Add the first!' on a new line is a good idea, but not sure about:

>the same font settings as the text in a comment

...because I believe across Calypso we make the distinction between user-generated content (always in a serif) and interface text (always in a sans-serif).

bluefuton commented:
All feedback addressed apart from 'add the first'. Thanks both!

fraying commented:
Honestly I'm not worried about that. It's clear from the context that "Add the first!" is not a user-posted comment. And I'm using SF UI Text for some things that are user voice in the refresh. 

Anyway, I'm happy to leave "Add the first!" in the header, but it should have a cap A because it's not a normal sentence, it's a header plus a sentence.

fraying commented:
@bluefuton Also, the links on the commenters are still opening in a new tab. They should open in the same tab, now that we've removed the "open in new" gridicon. Sorry I didn't make this clear originally.

bluefuton commented:
> Also, the links on the commenters are still opening in a new tab. They should open in the same tab, now that we've removed the "open in new" gridicon. 

Fixed 👍

fraying commented:
Thanks! This all LGTM!
